2019-08-03-q-and-as-about-nix-for-haskellers.md

```diff
@@ -0,0 +1,692 @@
+title = Questions and Answers about Nix for Haskellers
+tags = haskell, nixos
+summary = Overview of Nix for Haskell development
+draft = true
+
+------------------------------------------------------
+
+Last week Dmitrii Kovanikov ([@ChShersh](https://twitter.com/ChShersh)) posted
+a [request](https://twitter.com/ChShersh/status/1154782785967542272?s=20) on
+Twitter asking if anyone had the time to answer some questions about how to
+understand Nix as it applies to Haskell programming.
+
+Dmitrii is working on a scaffolding tool for Haskell projects called
+[summoner](https://github.com/kowainik/summoner).  There is an
+[open PR](https://github.com/kowainik/summoner/pull/283) adding Nix support.
+However, Dmitrii doesn't have any Nix experience, so he'd like to learn more
+about using Nix for Haskell development before tackling the PR.
+
+I've been using Nix for about a year and half, both at work and on my own
+projects.  I think there is a big lack of documentation surrounding Nix and
+Haskell, so I told him I'd be able to answer any questions he had.
+
+He wrote me back with a _great_ list of questions.  I thought it would be an
+injustice to the Haskell community if I didn't share my answers, so I decided
+to turn his questions and my answers into an article.
+
+This article is mainly for Haskellers that have heard about Nix (on Twitter,
+Reddit, YouTube, etc), know about its advantages, but haven't actually played
```

Comment:
Maybe it's a good idea not to assume that people know about `nix` advantages? This blog post tries to describe when people should and shouldn't not use nix, and this goes in a line with knowing about nix advantages. I wonder, whether it's possible to put `nix` advantages in a single simple sentence. Something like nix:

_If you configure your project with `nix` then you can build it on any machine with a single command without having anything preconfigured and achieve the same result each time._